pom.xml

```diff
@@ -66,33 +71,50 @@
           <target>${maven.compiler.target}</target>
         </configuration>
       </plugin>
-
       <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-checkstyle-plugin</artifactId>
-        <version>2.17</version>
+        <groupId>com.coveo</groupId>
+        <artifactId>fmt-maven-plugin</artifactId>
+        <version>2.5.1</version>
+        <executions>
+          <execution>
+            <goals>
+              <goal>check</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>com.mycila</groupId>
+        <artifactId>license-maven-plugin</artifactId>
+        <version>3.0</version>
+        <configuration>
+          <header>LICENSE.header</header>
+          <encoding>UTF-8</encoding>
+          <skipExistingHeaders>true</skipExistingHeaders>
+          <mapping>
+            <java>SLASHSTAR_STYLE</java>
+          </mapping>
+          <excludes>
+            <exclude>**/README.md</exclude>
+            <exclude>**/LICENSE</exclude>
+            <exclude>**/LICENSE.header</exclude>
+            <exclude>**/AUTHORS</exclude>
+            <exclude>**/*.xml</exclude>
+            <exclude>src/test/resources/**</exclude>
+            <exclude>src/main/resources/**</exclude>
```

Comment:
why exclude resources?

Reply:
should we modify the resources with license? in this case we can add a mapping resources like xml file and others.

Reply:
Well, i've heard some lawyers say that you don't need a header at all, as long as you have a license in the repo and in the META-INF or something.

The ones who tend to tell me that you should have a header on every file don't make exceptions... I'm not sure how we'd decide here.

Reply:
they do add the license header here though
https://github.com/google/closure-library/blob/master/closure/goog/testing/dom_test.html

Reply:
i will remove the resources exclusion, but should i also remove the AUTHORS and the xml exclusions too.

and i will appreciate a NOTICE file template.

Reply:
I don't think it is necessary to add a license header to the top of the AUTHORS file (usually just one at the root of the project, right?).

Reply:
Fwiw, never used it, but there's Spotless too that does both GJF and ensuring license headers: https://github.com/diffplug/spotless/tree/master/plugin-maven

Wrt AUTHORS, indeed it doesn't need a license header (it's also not _subject_ to the license, as it _defines_ who the copyright holders are).

Wrt NOTICE, we don't need it if we add the boilerplate notice to every file.